framework/src/positions/FunctorPositions.C

```diff
@@ -0,0 +1,75 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FunctorPositions.h"
+
+registerMooseObject("MooseApp", FunctorPositions);
+
+InputParameters
+FunctorPositions::validParams()
+{
+  InputParameters params = Positions::validParams();
+  params += NonADFunctorInterface::validParams();
+
+  params.addRequiredParam<std::vector<MooseFunctorName>>(
+      "positions_functors", "Functors providing the XYZ coordinates of the positions");
+  // Use user-provided ordering
+  params.set<bool>("auto_sort") = false;
+  // All functors defined on all processes for now
+  params.set<bool>("auto_broadcast") = false;
+  // Keep as up-to-date as possible given the generality of functors
+  params.set<ExecFlagEnum>("execute_on") = {EXEC_LINEAR, EXEC_TIMESTEP_BEGIN};
+
+  params.addClassDescription(
+      "Import positions from one or more reporters, for example other Positions");
+  return params;
+}
+
+FunctorPositions::FunctorPositions(const InputParameters & parameters)
+  : Positions(parameters), NonADFunctorInterface(this)
+
+{
+  const auto functor_names = getParam<std::vector<MooseFunctorName>>("positions_functors");
+
+  // Check input sizes
+  if (functor_names.size() % 3 != 0)
+    paramError("position_functors",
+               "The list of functors must be divisible by three, the number of coordinates");
+
+  for (const auto & name : functor_names)
+    _pos_functors.push_back(&getFunctor<Real>(name));
+
+  // Obtain the positions by evaluating the functors
+  initialize();
```

Comment:
During restart, do the positions get restarted or are they evaluated based on the initial evaluation of the functor?